pinot-core/src/main/java/com/linkedin/pinot/core/operator/BaseOperator.java

```diff
@@ -26,42 +25,25 @@
 /**
  * Any other Pinot Operators should extend BaseOperator
  */
-public abstract class BaseOperator implements Operator {
+public abstract class BaseOperator<T extends Block> implements Operator<T> {
   private static final Logger LOGGER = LoggerFactory.getLogger(BaseOperator.class);
 
   @Override
-  public final Block nextBlock() {
+  public final T nextBlock() {
     if (TraceContext.traceEnabled()) {
       long start = System.currentTimeMillis();
-      Block ret = getNextBlock();
+      T nextBlock = getNextBlock();
       long end = System.currentTimeMillis();
       String operatorName = getOperatorName();
       LOGGER.trace("Time spent in {}: {}", operatorName, (end - start));
       TraceContext.logTime(operatorName, (end - start));
-      return ret;
+      return nextBlock;
     } else {
       return getNextBlock();
     }
   }
 
-  @Override
-  public final Block nextBlock(BlockId blockId) {
-    if (TraceContext.traceEnabled()) {
-      long start = System.currentTimeMillis();
-      Block ret = getNextBlock(blockId);
-      long end = System.currentTimeMillis();
-      String operatorName = getOperatorName();
-      LOGGER.trace("Time spent in {}: {}", operatorName, (end - start));
-      TraceContext.logTime(operatorName, (end - start));
-      return ret;
-    } else {
-      return getNextBlock(blockId);
-    }
-  }
-
-  public abstract Block getNextBlock();
-
-  public abstract Block getNextBlock(BlockId blockId);
+  protected abstract T getNextBlock();
```

Comment:
Can we add the documentation on the reason why you put this as `protected`? I see that you put the explanation on the pr message but it would be helpful to put a documentation here.

Reply:
Added.